meta-intel-edison-bsp/recipes-support/gadget/files/conf-gadget.sh

```diff
@@ -6,14 +6,22 @@ set -euf -o pipefail
 readonly GADGET_BASE_DIR="/sys/kernel/config/usb_gadget/g1"
 readonly DEV_ETH_ADDR="aa:bb:cc:dd:ee:f1"
 readonly HOST_ETH_ADDR="aa:bb:cc:dd:ee:f2"
-readonly USBDISK="/dev/mmcblk0p9"
+readonly UPDATE_PART="/dev/mmcblk0p9"
+readonly BOOT_PART="/dev/mmcblk0p7"
 
 # Check if already run before
 if [ -d "${GADGET_BASE_DIR}" ]; then
     echo "Already registered gadgets"
     exit 0
 fi
 
+# Check if there is an update partition
+if [ -n "`ls /dev/disk/by-partlabel | grep update`" ]; then
```

Comment:
I guess it is easier to write like `if ... | grep -q ...; then` (i.o.w. no `test` is needed)

Reply:
I'll check this.

Reply:
I can change it to this one liner keeping the test but removing unnecessary processing:

`[ -h "/dev/disk/by-partlabel/update" ] && USBDISK=${UPDATE_PART} || USBDISK=${BOOT_PART}`